vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/patch_imagetemplate.go

```diff
@@ -114,24 +112,22 @@ type KeyFunc func(key string) (string, bool)
 
 // Versions is a KeyFunc for retrieving information about the current version.
 func Versions(key string) (string, bool) {
+	openshiftVersion := GetOpenshiftVersion()
```

Comment:
Comment here explaining that this will fail 100% of the time when called from `hyperkube kube-controller-manager --openshift-config` and provide the flags for the alternative.

In fact, you should probably give an information `glog.Fatal` if you don't have this information.

Reply:
I was thinking we should default to Kubernetes version defaults if we don't have this information.. but what alternative flag you are thinking?

Reply:
Alright added a fatal log that will log:

```
Error matching image format openshift/origin-${component}:${version} with Openshift version. 
Openshift can be additionally started with environment variable OPENSHIFT_RECYCLER_IMAGE to match image versions.
Expected no error while matching image versions - found the key "version" in "openshift/origin-${component}:${version}" is not recognized
```

If for some reason openshift version can not be found.